### DISCUSSION TITLE
Battery Lifetime

### DISCUSSION THREAD

**Discussion by Pferdebockwurst** [rbaron/b-parasite#179] [General]
Hi all,

I'm about to get a couple of b-parasites for my home. I have a pretty large ZigBee setup (108 devices right now) and so I want to go the ZigBee route with these sensors too. I try to avoid battery powered devices, but obviously there are cases like this, we're a mains powered device is not possible.

So what I'd like to know is, how is your experience with the battery life of the b-parasite. Preferably I hope for answers from other ZigBee users, but as there are probably not that many, Bluetooth users are also encouraged to share their battery lifetimes.

Also I'd really like to know if someone is using bigger coin batteries, like 2477 maybe. The manufacturer of the recommended battery holder for the 2032 also offers a nice holder for 2450 and [2477](https://www.lcsc.com/mobile/product-detail/Battery-Connectors_MYOUNG-MY-2477-02_C2979175.html), but the solder pads on the PCB obviously won't fit, because the ones on the holder are a couple of millimetres more apart. If someone has an idea how to get the bigger holders working, I'd be happy to know. Unfortunately I'm pretty now to this kicad stuff and so on, so I can't redesign the PCB on my own.

Best regards and thanks in advance.
Chris

**Comment by rbaron** [comment]
Hey @Pferdebockwurst,

I did some power consumption measurements in [this section in the Zigbee sample docs](https://github.com/rbaron/b-parasite/blob/main/code/nrf-connect/samples/zigbee/README.md#battery-life), which you can use to estimate battery life. The real world scenario depends on many factors, most importantly the configs listed [here](https://github.com/rbaron/b-parasite/blob/main/code/nrf-connect/samples/zigbee/README.md#configs).

**Comment by Pferdebockwurst** [comment]
Ok, thanks, I didn't see this. But I'm not really sure how to calculate the lifetime of the battery. Assuming the config is set so report the values every ten minutes, the average consumption per second is 2,205 uA. A 2032 coin cell has a capacity of 230 mAh. But I don't know how to go on from here 🫣

**Comment by rbaron** [comment]
You can plug those values in something like digikey's [battery life calculator](https://www.digikey.com/en/resources/conversion-calculators/conversion-calculator-battery-life).

These are very optimistic, so usually I'd expect between 1-2 years in practice.
